packages/edit-site/src/components/style-book/index.js

```diff
@@ -339,12 +344,26 @@ const StyleBookBody = ( {
 
 const Examples = memo(
 	( { className, examples, category, label, isSelected, onSelect } ) => {
-		const composite = useCompositeState( { orientation: 'vertical' } );
+		const compositeStore = useCompositeStore( {
+			orientation: 'vertical',
+			setItems: ( compositeItems ) => {
+				if ( ! compositeItems.length ) return;
+
+				for ( const compositeItem in compositeItems ) {
+					if ( compositeItem?.id === activeId ) return;
+				}
+
+				compositeStore.setActiveId( compositeItems[ 0 ]?.id );
+			},
```

Comment:
Out of curiosity, why is this code necessary? If it's for implementing an initial selection, have you considered using `defaultActiveId` ?

Reply:
This is because `compositeStore` seems to be shared across all the example tabs. If you interact with examples in one tab, then move to another tab, the `activeId` continues to point at an example which is no longer present. This means that none of the currently registered items are in the tab order, as they all have `tabindex="-1"`.

I might be missing something entirely that means we can contextualise `compositeStore`, of course. Or perhaps this might be better resolved at the `ariakit` level – if the item with the `activeId` is removed, the `activeId` might need to be reset. Wonder if @diegohaz has any insight.

Reply:
Is there a particular reason why all example tabs are wrapped in a single `Composite` element? If not, I recommend moving `Composite` into the `Example` component, allowing each example to maintain its own state.

Ariakit doesn't automatically update the `activeId` state when the current active item isn't mounted, as there could be several valid situations for this. For instance, the state might be preset in a virtualized list before the item element is mounted and scrolled into view, the element could mount/unmount rapidly when wrapped with Suspense, etc.

There are plans to create a separate component in Ariakit to deal with the disposal of composite items, but in my opinion, the aforementioned remains the best solution for this situation, assuming I understand the issue correctly.

Reply:
Oh, my apologies, I believe I misunderstood the component tree. `Example` is the row, and `Examples` represents a single tab, correct? If so, I'm puzzled as to why the state is shared across tabs. Perhaps something like `<Examples key={currentTab} />` might resolve the issue?

Reply:
> Example is the row, and Examples represents a single tab, correct?

Yes. `StyleBook` renders one or more `StyleBookBody` instances, each of which has its own `Examples` component. `Examples` renders an `Example` for every block.

Will give the `key` suggestion a go, and see what happens. Thanks.

Reply:
[Seems to have done the trick](https://github.com/WordPress/gutenberg/pull/55344/commits/c21998cd406b7d8cc65755e8c801eea382b99bcf). Thank you as always for the pointers!